### PR TITLE
Refactor/mocha to jest

### PR DESCRIPTION
**Pull request by sebbi08**

### CODE REVIEW

File: test/performance/Dockerfile [hunk 1]
```diff
@@ -2,4 +2,4 @@ FROM node:15.5.1
 
 COPY . .
 
-ENTRYPOINT npx mocha -r ts-node/register -r tsconfig-paths/register ./test/performance.ts --project ./test/tsconfig.json --exit
+ENTRYPOINT npx jest --testPathIgnorePatterns="[]" --testPathPattern="test/performance" --listTests
```
Comment (jkoenig134) [inline test/performance/Dockerfile:5]:
```suggestion
ENTRYPOINT npx jest --testPathIgnorePatterns="[]" --testPathPattern="test/performance"
```

File: package.json [hunk 2]
```diff
@@ -17,87 +17,95 @@
...
+    "jest": {
+        "maxWorkers": 5,
+        "preset": "ts-jest",
+        "setupFilesAfterEnv": [
+            "./test/customMatchers.ts",
+            "jest-expect-message"
+        ],
+        "testPathIgnorePatterns": [
+            "/test/performance/",
+            "/node_modules/"
```
Comment (jkoenig134) [inline package.json:53]: is this necessary?

File: test/performance/Dockerfile (hunk 1 above)
Comment (jkoenig134) [reply test/performance/Dockerfile:5]: this will exit without running the tests

File: test/testHelpers/AppDeviceTest.ts
Comment (jkoenig134) [inline test/testHelpers/AppDeviceTest.ts:1]: I think we don't need this file anymore

File: test/testHelpers/DeviceTestParameters.ts
Comment (jkoenig134) [inline test/testHelpers/DeviceTestParameters.ts:1]: is this still used?

File: test/testHelpers/FakeDbCollectionProvider.ts
Comment (jkoenig134) [inline test/testHelpers/FakeDbCollectionProvider.ts:1]: why is this removed? unused?

File: test/testHelpers/LogInterceptor.ts
Comment (jkoenig134) [inline test/testHelpers/LogInterceptor.ts:1]: unused?

File: test/testHelpers/MockDatabaseCollection.ts
Comment (jkoenig134) [inline test/testHelpers/MockDatabaseCollection.ts:1]: unused?

**Review by jkoenig134 — CHANGES_REQUESTED** [review]

File: test/modules/sync/BackboneConcurrency.test.ts [hunk 3]
```diff
@@ -1,97 +1,92 @@
...
 
-export class BackboneConcurrencyTest extends AbstractTest {
-    public constructor(config: IConfigOverwrite, connection: IDatabaseConnection, loggerFactory: ILoggerFactory) {
-        super({ ...config, datawalletEnabled: true }, connection, loggerFactory)
-    }
-
-    public run(): void {
-        const that = this
+    beforeEach(async () => {
+        // REVISIT is this ok ? multiple occurences in the sync tests
```
[the remark this answers is not in the document]
Comment (sebbi08) [reply test/modules/sync/BackboneConcurrency.test.ts:16]: the problem was i forgot to remove it :)

File: package.json (hunk 2 above)
Comment (sebbi08) [reply package.json:53]: it is the default and is overwritten when specified so i readded it in the end no test where found in node modules but better save than sorry

Comment (jkoenig134) [reply package.json:53]: ok, I thought this simply get added.

File: test/testHelpers/AppDeviceTest.ts
Comment (sebbi08) [reply test/testHelpers/AppDeviceTest.ts:1]: it is used in the DeviceOnboarding and DeviceDeletion test should i remove it and move the methods to the TestUtil ?

File: test/testHelpers/FakeDbCollectionProvider.ts
Comment (sebbi08) [reply test/testHelpers/FakeDbCollectionProvider.ts:1]: yes

File: test/testHelpers/LogInterceptor.ts
Comment (sebbi08) [reply test/testHelpers/LogInterceptor.ts:1]: y

File: test/testHelpers/MockDatabaseCollection.ts
Comment (sebbi08) [reply test/testHelpers/MockDatabaseCollection.ts:1]: y

File: test/testHelpers/AppDeviceTest.ts
Comment (jkoenig134) [reply test/testHelpers/AppDeviceTest.ts:1]: no, I was confused. It can stay

File: test/modules/sync/SyncController.callback.test.ts [hunk 4]
```diff
@@ -13,26 +12,27 @@ interface CallbackObject {
...
+
+    beforeAll(async () => {
+        connection = await TestUtil.createDatabaseConnection()
+        a1 = await TestUtil.createIdentityWithOneDevice(connection, { datawalletEnabled: true })
+        ;({ device1: b1, device2: b2 } = await TestUtil.createIdentityWithTwoDevices(connection, {
+            datawalletEnabled: true
+        }))
+    })
 
     afterEach(async function () {
```
Comment (jkoenig134) [inline test/modules/sync/SyncController.callback.test.ts:27]: afterAll, not afterEach

**Review by jkoenig134 — DISMISSED** [review]

**Review by jkoenig134 — APPROVED** [review]

**Pull request merged**